### DISCUSSION TITLE
Qodana Forward Reports

### DISCUSSION THREAD

**Discussion by Ne0nd0g** [JetBrains/Qodana#152] [Q&A] [answered]
I've successfully run Qodana against my project and reviewed the report locally. I set the `QODANA_TOKEN` to a token generated from my account and `QODANA_REMOTE_URL` to my project's GitHub URL. I don't see any forwarded reports to my account and project. How do I troubleshoot to see why the report isn't being forwarded?

**Comment by brichbash** [comment]
Hello,

Could you please share the command you run Qodana with and also logs from the latest Qodana run that failed to forward the report to Cloud?
You can access the logs by running `qodana show -d` if you're using [qodana-cli](https://github.com/JetBrains/qodana-cli).
If you're running directly via Docker, please add to the command `-v $(pwd)/results:/data/results/` and share the `log` directory. If you want to, you can send it to qodana-support@jetbrains.com

**Comment by Ne0nd0g** [comment, marked as answer]
I'm only setting the following environment variables from [here](https://www.jetbrains.com/help/qodana/cloud-forward-reports.html) on the host where qodana-cli is being run:

- `QODANA_TOKEN`
- `QODANA_REMOTE_URL`
- `QODANA_BRANCH`

Do these environment variables need to be set by me in the Docker container or does qodana-cli read them from the host where it is run?

I'm using the following command to run the scan `sudo /home/kali/go/bin/qodana-cli scan -l jetbrains/qodana-go:2022.3-eap`. I need to use sudo because Docker will not run without.

When I first ran a successful scan, I was doing so out of the temp directory, but my VM has since rebooted. I ran the following command but nothing was returned: `qodana-cli show -d --report-dir /root/.cache/JetBrains/jetbrains-qodana-go-2022.3-eap/d40ab877fd5d64a68e0ed5dd2d10b89a1db2aae3ec98ffdbd6340597e1db988b`

**Reply by brichbash** [reply]
The variables aren't read from the host. If you're running Qodana locally, you should also set the variables via `-e` flag, e.g: `qodana scan -e QODANA_TOKEN=$YOUR_TOKEN -e QODANA_BRANCH=main`. 
On GitHUb, you can add this to your Qodana job, and the variable will be used:
```
env:
    QODANA_TOKEN: ${{ secrets.QODANA_TOKEN }}
```

**Reply by Ne0nd0g** [reply]
I can't mark just your reply as the answer, only the first reply in the thread.

**Comment by Ne0nd0g** [comment]
Thank you, using the `-e` argument with `qodana-cli` works as expected and the report is in my dashboard.

I'm not using GitHub CI, I'm using AppVeyor which has a way to set environment variables, but I wouldn't want to use them on the command line because that information is exposed to the public.

**Comment by Ne0nd0g** [comment]
How do I find, or generate, a publicly accessible link to the cloud report?
